File: OneCodeTeam/How-to-bulk-update-the-HTML-field-eg-ReproSteps-27528/[VB]-How to bulk update the HTML field (e.g. ReproSteps) for TFS workitems/VB/VBTFSUpdateHTMLFieldsWorkItem/Common/Extensions.vb
'****************************** Module Header ******************************\
' Module Name:  Extensions.vb
' Project:      Common
' Copyright (c) Microsoft Corporation.
'
' Extensions class
'
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'***************************************************************************/

Imports Microsoft.TeamFoundation.WorkItemTracking.Client
Namespace Common


    Public NotInheritable Class Extensions
        Private Sub New()
        End Sub
#Region "Methods"


        Public Shared Function IsDisplayable(field As Field, revision As Integer) As Boolean
            If field.FieldDefinition Is Nothing Then
                Return False
            End If

            Select Case DirectCast(field.Id, CoreField)
                Case CoreField.History, CoreField.ChangedDate, CoreField.RevisedDate, CoreField.ChangedBy, CoreField.AuthorizedAs
                    Return False
                Case Else

                    Exit Select
            End Select

            If revision = 0 Then
                If field.Value Is Nothing OrElse Equals(field.Value, [String].Empty) OrElse Equals(field.Value, 0) Then
                    Return False
                End If
            End If

            Return True
        End Function

#End Region
    End Class
End Namespace

File: OneCodeTeam/How-to-bulk-update-the-HTML-field-eg-ReproSteps-27528/[VB]-How to bulk update the HTML field (e.g. ReproSteps) for TFS workitems/C#/CSTFSUpdateHTMLFieldsWorkItem/Helpers/Extensions.cs
/****************************** Module Header ******************************\
 * Module Name:  Extensions.cs
 * Project:      Common
 * Copyright (c) Microsoft Corporation.
 * 
 * Extensions class.
 * 
 * This source is subject to the Microsoft Public License.
 * See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL
 * All other rights reserved.
 * 
 * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
 * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
 * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\***************************************************************************/


namespace Common
{
    using System;

    using Microsoft.TeamFoundation.WorkItemTracking.Client;

    public static class Extensions
    {
        #region Methods

        public static bool IsDisplayable(this Field field, int revision)
        {
            if (field.FieldDefinition == null) return false;

            switch ((CoreField)field.Id)
            {
                case CoreField.History:
                case CoreField.ChangedDate:
                case CoreField.RevisedDate:
                case CoreField.ChangedBy:
                case CoreField.AuthorizedAs:
                    return false;

                default:
                    break;
            }

            if (revision == 0)
            {
                if (field.Value == null ||
                   Equals(field.Value, String.Empty) ||
                   Equals(field.Value, 0)) return false;
            }

            return true;
        }

        #endregion Methods
    }
}